matchbox_socket/src/webrtc_socket/native/message_loop.rs

```diff
@@ -71,14 +72,22 @@ async fn message_loop_impl(
     let mut peer_loops_a = FuturesUnordered::new();
     let mut peer_loops_b = FuturesUnordered::new();
     let mut handshake_signals = HashMap::new();
-    let mut connected_peers = HashMap::new();
+    let mut connected_peers: HashMap<String, Vec<UnboundedSender<Box<[u8]>>>> = HashMap::new();
 
     let timeout = Delay::new(Duration::from_millis(KEEP_ALIVE_INTERVAL));
     futures::pin_mut!(timeout);
 
     loop {
         let next_signal_event = events_receiver.next().fuse();
-        let next_peer_message_out = peer_messages_out_rx.next().fuse();
+
+        let mut next_peer_messages_out = FuturesUnordered::new();
+        peer_messages_out_rx
+            .iter_mut()
+            .enumerate()
+            .map(|(index, r)| async move { (index, r.next().fuse().await) })
+            .for_each(|f| next_peer_messages_out.push(f));
+
+        let next_peer_message_out = next_peer_messages_out.next().fuse();
```

Comment:
I just imitated the way it was before but I also think moving this to the select branches is more readable

Reply:
Ok, feel free to leave it as is.

I really wish properly fusing futures in rust wasn't as footgun-ey :/

Reply:
I have to confess I don't really understand the problem and I am also not sure about the relevance of fusing the future in this particular place because I believe these futures only return at most once (because they are just `select!`ed once) but it's very possible I am wrong...

Reply:
Yes, these concepts keep slipping out of my mind when it's a long time since I looked at them :/

As far as I understand the documentation for select, it may keep polling the same future multiple times, so they have to be fused in order not to keep polling finished futures.

https://rust-lang.github.io/async-book/06_multiple_futures/03_select.html

> Futures and streams which are not already fused can be fused using the `.fuse()` method. Note, though, that fusing a future or stream directly in the call to `select!` will not be enough to prevent it from being polled after completion if the `select!` call is in a loop, so when `select!`ing in a loop, users should take care to `fuse()` outside of the loop.

https://docs.rs/futures/latest/futures/macro.select.html

I think maybe the right thing to do here would be to fuse the streams instead (if they're not already implementing FusedStream).

Will have to look more into this later, out of time now.

Reply:
The mpsc channels implement `FusedStream`, so i think we at least can get rid of the `.next().fuse()`